### DISCUSSION TITLE
Issues with .mat result file and big file size

### DISCUSSION THREAD

**Discussion by hillbu1** [OpenModelica/OpenModelica#12524] [General]
Hello,

Currently we are facing issues with result files as .mat having a big file size of 2 - 3 GB or more. Then the file can't be opened. Matlab  states `Unknown text on line number 1 of ASCII file`, also Python is not able to open that file (error message using scipy.loadmat: `ValueError: read length must be non-negative or -1`).
If we reduce the file size by shortening the simulated time span the file works. 
First, that issue occured on my notebook and we thought that it might be due limited memory, so we moved to a workstation having 192 GB of memory. But still the issue is present, i.e., memory seems not to be the issue.
For now, we found a workaround to split the simulation into several parts with some overlapping and combine the results afterwards in the postprocessing. Unfortunately, for some applications we would need to forward the results from one part of the simulation to the next one, which would make things much more complicated.

Does anybody of you have an idea what might be the root cause for that .mat file issue and how to overcome it?

**Comment by casella** [comment]
@hillbu1 we solved this problem in OMEdit already, by using fseek64() instead of fseek(), see #10706. However, I guess this is not according to the .mat v4 standard.

What is exactly the status with Matlab .mat file sizes is not too clear to me. [This post](https://it.mathworks.com/matlabcentral/answers/1903065-what-is-the-maximum-size-of-mat-file) on the official Matlab site says there's no limit to .mat files. This other [official page](https://it.mathworks.com/help/matlab/import_export/mat-file-versions.html) seems to hint that in order to go above 2GB you need to save files with v7.3, which is an HDF5 compressed format, much more involved than previous standards as .mat v4.

The problem is, these Modelica output files use a de-facto standard set by Dymola which has never been standardized. So, it's a bit messy. Yes, we need to fix this problem, but we should do it with the Modelica Association.

For the time being, maybe you can use the variable filter and only save relevant variables. You hardly really need all of those GB of data, do you?

**Reply by hillbu1** [reply]
Hello @casella,

thanks for your reply and sorry for my delayed answer, I was out of office for two weeks.
I already feared that this might be a more general topic that might not be easy to solve.
Actually we already reduced the output variables drastically with the variable filter. And we're also using `-single` for single precision and `-noEventEmit` to ensure a certain time interval, currently 1 s. So I think the time interval is the only paramter we could play with at the moment to reduce the file size any further.
